test/extended/README.md

```diff
@@ -0,0 +1,145 @@
+
+The structure of this directory is following:
+
+* [**`test/extended/util`**](util) provides useful helpers and utilities to use in your extended test. It provides a easy-to-use interface to OpenShift CLI and also
+access to the Kubernetes [E2E framework](https://github.com/openshift/origin/tree/master/Godeps/_workspace/src/k8s.io/kubernetes/test/e2e) helpers. It also contains OpenShift helpers are shared across multiple test cases, to make the test cases more DRY.
+* [**`test/extended/fixtures`**](fixtures) contains the JSON and YAML fixtures are meant to be used by the extended tests.
+* [**`test/extended/[images,builds,...]`**](builds) each of this Go packages contains extended tests that are related together. For example, the `images` directory should contain test cases that are exercising usage of various Docker images in OpenShift.
+* [**`hack/test-extended/[group]/run.sh`**](../../hack/test-extended) contains a shell script that helps you to execute extended tests for a certain group
+* [**`test/extended/extended_test.go`**](extended_test.go) is a runner for all extended test packages. Looking inside this file, you can see how you can add new extended test Go package to be compiled:
+```go
+	_ "github.com/openshift/origin/test/extended/builds"
+	_ "github.com/openshift/origin/test/extended/images"
+```
+
+Groups vs. packages
+---------------------
+
+Since the extended tests might rely on specify OpenShift server configuration,
+the tests are divided into logical 'test groups'. Each group has its own shell
+launcher that bootstraps the OpenShift environment in a way the group requires
+to be executed.
+For example, you might want to write an extended test for the LDAP
+authentication which means that you have to configure OpenShift server to
+enable this authentication method. 
+You can create a new test group `ldap` and provide a shell launcher
+`./hack/test-extended/ldap/run.sh` to start OpenShift with required
+configuration.
+Then you place the source code with extended test into the extended test Go
+package that corresponds to functionality you are going to test. In case of
+LDAP, it can be `./test/extended/authentication`.  In order to have your test
+case executed by `./hack/test-extended/ldap/run.sh` you have to add `ldap:`
+prefix to the `Describe()` function:
+
+Example:
+```go
+var _ = g.Describe("ldap: Authenticate using LDAP", func() {
+  # ...
+})
+```
+
+
+CLI interface
+---------------------
+
+In order to be able to call the OpenShift CLI and Kubernetes and OpenShift REST client and simulate the OpenShift `oc` command in the test suit, first we need to create an instance of the CLI, in the top-level Ginkgo describe container. The top-level describe container shall also specify the the bucket, into which the test belongs and short test description. Other globally accessible variables can be declared(eg. fixtures) can be declared as well.
```

Comment:
s/ client / clients /
s/ test suit / test suite /
s/ The top-level describe container shall also specify the the bucket, into which the test belongs and short test description. Other globally accessible variables can be declared(eg. fixtures) can be declared as well. / The top-level describe container should also specify the buck into which the test belongs and a short test description.  Other globally accessible variables (eg. fixtures) can be declared as well /